apps/utils/locust_tests/common/stats/stats_compare.py

```diff
@@ -1,212 +1,332 @@
+        given Locust Task
 
     Returns:
-        List[StatCompareResult]: A filtered list of comparison results that exceed the threshold percentage given
+            List[StatComparison]: A list of comparison results for each TasksStats stats
     """
-    return [stat_delta for stat_delta in compare_results if stat_delta.percent > threshold]
+    prev_stats = {**asdict(previous), **previous.response_time_percentiles}
+    cur_stats = {**asdict(current), **current.response_time_percentiles}
+    # Compute the list of StatCompareResults for each type of field
+    return _compute_results_list(prev_stats, cur_stats, stats_compare_metadata)
 
 
 def validate_aggregated_stats(
-    previous: AggregatedStats,
-    current: AggregatedStats,
-    threshold: float = DEFAULT_DEVIANCE_FAILURE_THRESHOLD,
-) -> Dict[str, List[StatCompareResult]]:
-    """Validates and compares the given AggregatedStats instances against each other, checking each of their common
-    TaskStats and returning a dictionary of the name of those tasks that exceed the given threshold to the actual stats
-    that failed
+    previous: AggregatedStats, current: AggregatedStats, all_comparisons_meta: AllComparisonMetadata
+) -> Tuple[List[StatComparison], Dict[str, List[StatComparison]]]:
+    """Validates and compares the given AggregatedStats instances against each other, checking each
+    of their common TaskStats and returning a dictionary of the name of those tasks that exceed the
+    given threshold to the actual stats that failed
 
     Args:
-        previous (AggregatedStats): A previous run or average of all previous runs that will be compared against for the curren run
-        current (AggregatedStats): The current run's stats
-        threshold (float, optional): A percent threshold that the current run's task's stats must not exceed/be worse than. Defaults to DEFAULT_PERCENT_THRESHOLD
+            previous (AggregatedStats): A previous run or average of all previous runs that will
+            be compared against for the current run
+            current (AggregatedStats): The current run's stats
+            all_comparisons_meta (AllComparisonMetadata): All of the stat comparison metadata needed
+            to compare stats
 
     Returns:
-        Dict[str, List[StatCompareResult]]: A dictionary of failing task names to the stats that failed along with their comparison results
+            Tuple[List[StatComparison], Dict[str, List[StatComparison]]]: A Tuple where the first
+            entry is the list of comparison results for the aggregated totls and the second entry
+            is a dictionary of tasks to their list of comparison results
     """
     prev_tasks = {task.task_name: task for task in previous.tasks}
     cur_tasks = {task.task_name: task for task in current.tasks}
     common_tasks = prev_tasks.keys() & cur_tasks.keys()
 
-    failed_tasks_with_percents = {}
+    tasks_exceeded_results = {}
```

Comment:
this name isn't immediately clear on what it is; could this be named to be slightly clearer? (I think calling what I think are the results or runs "tasks" is throwing me a bit when the documentation references results)

Reply:
So `tasks_exceeded_results` is short for "all `StatCompareResult`s that exceeded the passed-in `threshold`% for each Locust `Task`s keyed by the `Task`'s name". Whenever I use "task" in a Locust context, I am talking about a single test case within a test suite (Locustfile), i.e.:

```python
@tag("coverage", "coverage_test_id_count")
@task
def coverage_test_id_count(self):
    """Coverage search by ID, Paginated"""
    self.run_task_by_parameters(
        base_path="/v1/fhir/Coverage",
        params={"beneficiary": self.bene_ids.pop(), "_count": "10"},
        name="/v1/fhir/Coverage search by id / count=10",
    )
```

Does that help with your understanding? I do agree that this particular naming is probably a bit too terse or difficult to understand without the context I have from writing it. I'll think about a different name and/or adding a comment that matches what I wrote above.

Reply:
Yup that helps! If they are essentially test case results I'd have preferred they be named something that indicates that, but not going to ask you go and adjust the whole codebase over it. I think just adjusting the variable names like you suggested here would go a long way in describing it to the uninitiated!